lms/djangoapps/certificates/management/commands/cert_whitelist.py

```diff
@@ -12,6 +12,18 @@
 from django.contrib.auth.models import User
 
 
+def get_user_from_identifier(identifier):
+    """
+     This function takes the string identifier and fetch relevant user object from database
+    """
+    identifier = identifier.strip()
+    if '@' in identifier:
```

Comment:
This seems like a hack way of detecting the e-mail address.  Right now we don't allow non-alphanumeric characters in usernames but that may not always be the case.  I think this logic should try to get the the user by username first and if that fails, it should try to use the identifier as an e-mail address and find the user by e-mail address.